### PR TITLE
Add support to keep tracks of changes to fields.

### PR DESCRIPTION
**Pull request by ffaraoneim**
Add sphinx documentation.
Fix version of dependencies.

Fix F expressions for updates.
Fix for model private fields (GenericForeignKey).

Add integration tests for:

* Master: current -> Replica: v1.3.1
* Master: v1.3.1 -> Replica: current

### CODE REVIEW

File: dj_cqrs/constants.py [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 ALL_BASIC_FIELDS = '__all__'
 
+FIELDS_TRACKER_FIELD_NAME = '__fields_tracker'
+TRACKED_FIELDS_ATTR_NAME = '__tracked_fields'
 
-#  Copyright © 2020 Ingram Micro Inc. All rights reserved.
 
 class SignalType:
-    DELETE = 'DELETE'
+    """Type of signal that generates this event."""
```
Comment (maxipavlovic) [inline dj_cqrs/constants.py:10]: Strange text for DELETE signal.

File: dj_cqrs/transport/rabbit_mq.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def produce(cls, payload):
     def _consume_message(cls, ch, method, properties, body):
         try:
             dct = ujson.loads(body)
-            for key in ('signal_type', 'cqrs_id', 'instance_data'):
+            for key in ('signal_type', 'cqrs_id', 'instance_data', 'tracked_fields_data'):
```
Comment (maxipavlovic) [inline dj_cqrs/transport/rabbit_mq.py:69]: The more I think on that, the less I see the reasons to break backward compatibility. Consider the case, when there will be no downtime for upgrade, so there may be loses of important updates.

I guess, the support for that field should be optional by nature. If you don't want to deal with changes, you could have the transport that just ignores them, for example.

**Comment by ffaraoneim** [comment]
> Awesome changes! I love the docs, thank you very much.
> 
> One general question: did you run integration tests locally to see if nothing has broken? Moreover, it would be good to have one integration test, related to field tracking, added to the codebase.

Yes, integration tests pass but sure, I will add some tests related to tracking.

File: dj_cqrs/dataclasses.py [hunk 3]
```diff
@@ -29,10 +48,21 @@ def pk(self):
...
+        Return the payload as a dictionary.
+
+        :return: This payload.
+        :rtype: dict
+        """
         return {
             'signal_type': self.__signal_type,
             'cqrs_id': self.__cqrs_id,
             'instance_data': self.__instance_data,
+            'tracked_fields_data': self.__tracked_fields_data,
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply dj_cqrs/dataclasses.py:66]: I think instance_data and previous_data could be fine

File: requirements/dev.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
 Django >= 1.11.20
-pika
-ujson
\ No newline at end of file
+pika==1.1.0
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply requirements/dev.txt:2]: Sure I will put >=

File: dj_cqrs/metas.py [hunk 5]
```diff
@@ -22,13 +24,38 @@ def __new__(mcs, *args):
...
+
         if model_cls.CQRS_SERIALIZER is None:
             MasterMeta._check_cqrs_fields(model_cls)
 
         MasterRegistry.register_model(model_cls)
         MasterSignals.register_model(model_cls)
         return model_cls
 
+    @staticmethod
+    def _check_cqrs_tracked_fields(model_cls):
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply dj_cqrs/metas.py:40]: FK and 1to1 are supported and I think they could be useful. M2M are not supported by FieldTracker from model_utils they are simply ignored. We can assert that there is no m2m field declared in CQRS_TRACKED_FIELDS

File: dj_cqrs/managers.py [hunk 6]
```diff
@@ -47,20 +48,34 @@ def save_instance(self, master_data, sync=False):
...
 
         :param dict mapped_data: Mapped CQRS master instance data.
+        :param dict tracked_fields_data: Previous values for tracked fields.
         :param bool sync: Sync package flag.
         :return: ReplicaMixin model instance.
         :rtype: django.db.models.Model
         """
         try:
-            return self.model.cqrs_create(sync, **mapped_data)
+            return self.model.cqrs_create(
```
[the remark this answers is not in the document]
Comment (ffaraoneim) [reply dj_cqrs/managers.py:74]: Sure!

**Pull request merged**